test/libsolidity/syntaxTests/structs/recursion/recursive_struct_nested_mapping_storage.sol

```diff
@@ -0,0 +1,6 @@
+library a {
+    struct b {
+        mapping (uint => b) c ;
```

Comment:
Remind me, why do we allow mappings in structs?

Reply:
If you want to see an example of it being used, then
  https://github.com/JoinColony/colonyNetwork/blob/5cd0c042e270b665a49d6d024846a1e4bf4829fc/contracts/colony/ColonyDataTypes.sol#L274

Reply:
However, there are way too many bugs related to allowing mappings inside structs and arrays. So this is a tricky feature.

Reply:
And these are still valid:
```
contract C {
    struct S {
        mapping (uint => uint) m;
    }
    S s1;
    mapping (uint => uint)[] a;
    function f() public {
        delete s1;
        delete a;
        a.pop();
    }
}
```

Reply:
Yes, there is an open issue about `delete` and `pop` for data structures that contain mappings.